sympy/printing/glsl.py

```diff
@@ -400,6 +427,32 @@ def glsl_code(expr,assign_to=None,**settings):
        6, 7, 8, 9, 10
     ) /* a 2x5 matrix */
 
+    Any object printed by default as a GLSL float array can be printed with an
+    alternative constructor by passing a template string via the ``array_constructor``
+    parameter:
+    >>> int_constructor = 'int[${size}]'
+    >>> glsl_code(Matrix([1,2,3,4,5]), array_constructor=int_constructor)
+    'int[5](1, 2, 3, 4, 5)'
```

Comment:
Why not just make the argument `int`? Aren't the square brackets always required?

Reply:
Hey, this was really good feedback.  I got rid of the `array_constructor` param and replaced it with a `base_type` parameter, as you've suggested.

Your insight to use a custom class is awesome... I'll have to try that soon!